scripts/release/other_identifiers/retrieve_indirectIdentifiers_from_mart.pl

```diff
@@ -1,77 +1,160 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl
 use strict;
+use warnings;
 
 use lib "/usr/local/reactomes/Reactome/development/GKB/modules";
 
 use GKB::Config;
-use GKB::DBAdaptor;
-use GKB::EnsEMBLMartUtils qw/:all/;
-use GKB::Utils;
+use GKB::Config_Species;
+use GKB::EnsEMBLMartUtils qw/get_identifiers get_species_mart_name get_wget_query_for_identifier/;
 
-use Data::Dumper;
+use Capture::Tiny qw/capture/;
 use Getopt::Long;
+use List::MoreUtils qw/any/;
 use Try::Tiny;
 
 use Log::Log4perl qw/get_logger/;
-Log::Log4perl->init(\$LOG_CONF);
-my $logger = get_logger(__PACKAGE__);
-
-our($opt_user,$opt_host,$opt_pass,$opt_port,$opt_db,$opt_debug,$opt_sp);
-
-(@ARGV) || die "Usage: $0 -sp 'species name'
--user db_user -host db_host -pass db_pass -port db_port -db db_name\n";
-
-&GetOptions("user:s", "host:s", "pass:s", "port:i", "db:s", "debug", "sp=s");
-
-$opt_db || die "Need database name (-db).\n";    
-
-# Get connection to reactome db
-my $dba = GKB::DBAdaptor->new
-    (
-     -user   => $opt_user,
-     -host   => $opt_host,
-     -pass   => $opt_pass,
-     -port   => $opt_port,
-     -dbname => $opt_db,
-#     -DEBUG => $opt_debug
-     );
-
-# Fetch the species
-$opt_sp ||= 'Homo sapiens';
-
-my $sp = $dba->fetch_instance_by_attribute('Species',[['name',[$opt_sp]]])->[0]
-    || $logger->error_die("No species '$opt_sp' found.\n");
-my $sp_mart_name;
-if ($sp->displayName =~ /^(\w)\w+ (\w+)$/) {
-    $sp_mart_name = lc("$1$2");
-} else {
-    $logger->error_die("Can't form species abbreviation for mart from '" . $sp->displayName . "'.\n");
+
+# This statement allows the script either to be executed or loaded as a module
+# for testing purposes.
+# https://www.perl.com/article/107/2014/8/7/Rescue-legacy-code-with-modulinos/
+run(@ARGV) if !caller();

```

Comment:
Would it not work in this form?

```perl
if !caller()
{
    run(@ARGV);
}
```

Reply:
It would but I think the accepted practice is with the trailing if.  The article that discusses it and other places I've seen use it all do it that way.

Reply:
I see your point, though.  I think I will change it to use an if-block.

Reply:
Ok. I still think the trailing logic expressions (if/unless) are awful, but if that's considered to be the best practice in this context, I'll leave it alone.

Reply:
It may be the accepted practice for this particular usage, but I don't think it loses anything going to a full if block and it would be clearer for future readers.